jobbergate-cli/jobbergate_cli/subapps/job_scripts/tools.py

```diff
@@ -180,6 +188,146 @@ def get_template_output_name_mapping(config: JobbergateConfig, job_name: str) ->
+
+    render_contexts = [parameters, {"data": parameters}]
+
+    for context in render_contexts:
+        try:
+            return template.render(**context)
+        except UndefinedError as e:
+            logger.debug(
+                "Unable to render filename={} with context={} -- Error: {}",
+                template_path,
+                context,
+                str(e),
+            )
+
+    raise Abort(
+        f"Unable to render filename={template} with context={context}",
+        subject="Unable to render jinja template",
+        log_message=f"Unable to render filename={template} with context={context}",
+    )
+
+
+def render_job_script_locally(
+    jg_ctx: JobbergateContext,
+    job_script_name: str,
+    application_path: pathlib.Path,
+    output_path: pathlib.Path,
+    sbatch_params: Optional[List[str]] = None,
+    param_file: Optional[pathlib.Path] = None,
+    fast: bool = False,
+):
+    """
+    Render a new job script from an application in a local directory.
+
+    :param str job_script_name: Name of the new job script.
+    :param pathlib.Path application_path: Path to the base application.
+    :param pathlib.Path output_path: Path to the output the rendered job script.
+    :param Optional[List[str]] sbatch_params: List of sbatch parameters.
+    :param Optional[pathlib.Path] param_file: Path to a parameters file.
+    :param bool fast: Whether to use default answers (when available) instead of asking the user.
+    :param JobbergateContext jg_ctx: The Jobbergate context.
+    :return JobScriptResponse: The new job script.
+    """
+    # Make static type checkers happy
+    assert jg_ctx.client is not None
+
+    app_data = fetch_application_data_locally(application_path)
+
+    if not app_data.workflow_files:
+        raise Abort(
+            "Application does not have a workflow file",
+            subject="Workflow file not found",
+            log_message="Application does not have a workflow file",
+        )
+
+    application_source_code = app_data.workflow_files[0].path.read_text()
+
+    (app_config, app_module) = load_application_data(app_data, application_source_code)
+
+    supplied_params = validate_parameter_file(param_file) if param_file else dict()
+    execute_application(app_module, app_config, supplied_params, fast_mode=fast)
+    update_template_files_information(app_data, app_config)
+
+    param_dict_flat = flatten_param_dict(app_config.dict())
+
+    template_name_mapping = get_template_output_name_mapping(app_config.jobbergate_config, job_script_name)
+
+    mapped_template_files = {
+        new_filename: file
+        for file in app_data.template_files
+        if (new_filename := template_name_mapping.get(file.filename, None))
+    }
+
+    for new_filename, template_file in mapped_template_files.items():
+        file_content = render_template(template_file.path, param_dict_flat)
+
+        if template_file.file_type.upper() == "ENTRYPOINT" and sbatch_params:
```

Comment:
Why not use the enum here and on the schemas?

Reply:
I based the `LocalTemplateFile` on the `TemplateFileResponse` and it's using `str` there instead of the enum, so I overlooked it.
But that's a good catch, I updated to use the `enum` here.